nilearn/glm/first_level/first_level.py

```diff
@@ -1060,6 +1034,249 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
+    # as well as the ses field if more than one session is present.
+    msg_end = ('Please verify that the desc_label and space_label labels '
+               'corresponding to the BIDS spec were correctly specified.')
+    run_check_list = []
+    for img in imgs:
+        img_dict = parse_bids_filename(img)
+        if (
+            '_ses-' in img_dict['file_basename']
+            and '_run-' in img_dict['file_basename']
+        ):
+            if (img_dict['ses'], img_dict['run']) in run_check_list:
+                raise ValueError(
+                    'More than one nifti image found '
+                    f"for the same run {img_dict['run']} and ",
+                    f"session {img_dict['ses']}. {msg_end}")
+            else:
+                run_check_list.append((img_dict['ses'], img_dict['run']))
+
+        elif '_ses-' in img_dict['file_basename']:
+            if img_dict['ses'] in run_check_list:
+                raise ValueError(
+                    'More than one nifti image ',
+                    f"found for the same ses {img_dict['ses']}, while ",
+                    f'no additional run specification present. {msg_end}')
+            else:
+                run_check_list.append(img_dict['ses'])
+
+        elif '_run-' in img_dict['file_basename']:
+            if img_dict['run'] in run_check_list:
+                raise ValueError(
+                    'More than one nifti image '
+                    f"found for the same run {img_dict['run']}. {msg_end}")
+            else:
+                run_check_list.append(img_dict['run'])
+
+
+def _bids_check_events_list(events: list[str] | None,
+                           imgs: list[str],
+                           sub_label: str,
+                           task_label: str,
+                           dataset_path: str,
+                           events_filters: list[str],
+                           supported_filters: list[str]) -> None:
+    """_summary_
+
+    Parameters
+    ----------
+    events : list[str] | None
+        _description_
+    imgs : list[str]
+        _description_
+    sub_label : str
+        _description_
+    task_label : str
+        _description_
+    dataset_path : str
+        _description_
+    events_filters : list[str]
+        _description_
+    supported_filters : list[str]
+        _description_
+
+    Raises
+    ------
+    ValueError
+        _description_
+    """    
+    if not events:
+        raise ValueError('No events.tsv files found '
+                            f'for subject {sub_label} '
+                            f'for filter: {events_filters}.')
+    if len(events) != len(imgs):
+        raise ValueError(f'{len(events)} events.tsv files found'
+                            f' for {len(imgs)} bold files. '
+                            'Same number of event files '
+                            'as the number of runs is expected.')
+    for this_img in imgs:
```

Comment:
> I think you should just use `images` and `image` instead of `img`. There's no real benefit to saving 2 characters, esp with autocomplete. 

I think I had a quick look at the rest of the code base to see and it seems that `imgs` and `img` were used more often. 
So I tried to go for consistency but I do prefer full name as much as possible.

> This notation of using an `_` at the end of the singular in a loop is often indicative of iterator, and reduces chances of mistyping.

Did not know that one. Is that a common convention in python?

Reply:
> So I tried to go for consistency but I do prefer full name as much as possible.

C'est bon.

 > Did not know that one. Is that a common convention in python?

It is not canon, but IIRC, Raymond Hettinger mentioned this in one of his "How to write better Python code" talks, & it hs served me beautifully.